File: packages/api/src/Domain/Event.cs
namespace Domain
{
    public class Event
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public bool IsCancelled { get; set; }
        public ICollection<EventAttendee> Attendees { get; set; } = new List<EventAttendee>();
        public ICollection<Comment> Comments { get; set; } = new List<Comment>();
    }
}

File: packages/client/src/types/event.ts
import { Profile } from "./profile";

export interface Event {
  id: string;
  title: string;
  description: string;
  startDate: Date;
  endDate: Date;
  hostUsername: string;
  isCancelled: boolean;
  isGoing: boolean;
  isHost: boolean;
  host?: Profile;
  attendees: Profile[];
}

export class Event implements Event {
  constructor(init?: EventFormValues) {
    Object.assign(this, init);
  }
}

export class EventFormValues {
  id?: string = undefined;
  title: string = "";
  description: string = "";
  startDate: Date = new Date();
  endDate: Date = new Date();

  constructor(event?: EventFormValues) {
    if (event) {
      this.id = event.id;
      this.title = event.title;
      this.description = event.description;
      this.startDate = event.startDate;
      this.endDate = event.endDate;
    }
  }
}
